File: src/app/models/service-type.ts
/*
 * Copyright (C) 2019 The "mysteriumnetwork/mysterium-vpn-mobile" Authors.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

export enum ServiceType {
  Wireguard = 'wireguard',
  Openvpn = 'openvpn'
}

export function stringToServiceType(string: string): ServiceType | null {
  for (let key in ServiceType) {
    const value = ServiceType[key]
    if (string === value) {
      return value as ServiceType
    }
  }
  return null
}

File: android/app/src/main/java/updated/mysterium/vpn/model/proposal/parameters/ServiceType.kt
package updated.mysterium.vpn.model.proposal.parameters

enum class ServiceType(val type: String) {
    UNKNOWN("unknown"),
    OPENVPN("openvpn"),
    WIREGUARD("wireguard");

    companion object {
        fun parse(type: String): ServiceType {
            return values().find { it.type == type } ?: UNKNOWN
        }
    }
}
